app/src/main/java/ai/elimu/kukariri/service/ScreenOnService.kt

```diff
@@ -16,9 +19,18 @@ class ScreenOnService : Service() {
         Log.i(javaClass.name, "onStartCommand")
 
         // Register receiver for detecting when the screen is turned on
-        val screenOnReceiver = ScreenOnReceiver()
+        screenOnReceiver = ScreenOnReceiver()
         registerReceiver(screenOnReceiver, IntentFilter(Intent.ACTION_SCREEN_ON))
 
         return super.onStartCommand(intent, flags, startId)
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        try {
+            unregisterReceiver(screenOnReceiver)
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
@tuancoltech Use `Log` instead here. Something link Log.e(...)?

Reply:
@jo-elimu Sure. It's updated.